### PR TITLE
(google) Calculate and cache instance type capabilities when configur…

### PR DESCRIPTION
**Pull request by duftler**
…ing account.
(google) Return instance type details with credentials details.
(google) Verify zone and region in validation phase.
(google) Cut down on live calls to verify zone, region and instance type.

### CODE REVIEW

**Comment by duftler** [comment]
@danielpeach please review.

**Comment by duftler** [comment]
The motivation for these changes is that different projects may have different regions/zones/instanceTypes. Would like to try to dynamically determine what's available to each project and propagate those constraints to the ui where reasonable.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/validators/StandardGceAttributeValidator.groovy [hunk 1]
```diff
@@ -322,15 +300,19 @@ class StandardGceAttributeValidator {
     }
 
     if (location) {
-      def knownLocation = location in vCpuMaxByLocation
+      def knownLocation = location in credentials.locationToInstanceTypesMap
 
       // since our list might be out of date, we'll assume a location can handle at least 16 vCpus per machine.
       if (!knownLocation && vCpuCount > 16) {
```
Comment (danielpeach) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/validators/StandardGceAttributeValidator.groovy]: Should this logic still exist? Or is the `locationToInstanceTypesMap` not definitive?

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/security/GoogleNamedAccountCredentials.groovy [hunk 2]
```diff
@@ -110,13 +113,20 @@ class GoogleNamedAccountCredentials implements AccountCredentials<GoogleCredenti
...
       return this
     }
 
+    @VisibleForTesting
+    Builder locationToInstanceTypesMap(Map<String, Map> locationToInstanceTypesMap) {
+      this.locationToInstanceTypesMap = locationToInstanceTypesMap
+      return this
+    }
+
     Builder regionLookupEnabled(boolean enabled) {
```
Comment (danielpeach) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/security/GoogleNamedAccountCredentials.groovy]: Should this now be `liveLookupsEnabled`?

**Comment by danielpeach** [comment]
LGTM, minor comments

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/validators/StandardGceAttributeValidator.groovy (hunk 1 above)
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/validators/StandardGceAttributeValidator.groovy]: Good point. I've removed it.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/security/GoogleNamedAccountCredentials.groovy (hunk 2 above)
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/security/GoogleNamedAccountCredentials.groovy]: Done.

**Pull request merged**